pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -100,6 +102,22 @@ def __verify_eval_loop_configuration(self, model: 'pl.LightningModule', stage: s
             rank_zero_warn(f'you passed in a {loader_name} but have no {step_name}. Skipping {stage} loop')
         if has_step and not has_loader:
             rank_zero_warn(f'you defined a {step_name} but have no {loader_name}. Skipping {stage} loop')
+        if has_step and has_loader:
+            sig_args = inspect.signature(getattr(model, step_name)).parameters
+            loaders = getattr(model, loader_name)()
+            len_loaders = 1 if not isinstance(loaders, list) else len(loaders)
+            if (len(sig_args) == 2 and len_loaders != 1):
+                raise MisconfigurationException(
+                    f'You provide multiple {stage} dataloaders, but no `dataloader_idx` argument'
+                    f' given in {step_name}.'
+                )
+            if (len(sig_args) == 3
+                and len_loaders < 2) and sig_args['dataloader_idx'].default is not inspect.Parameter.empty:
```

Comment:
```suggestion
            if len(sig_args) == 3 and len_loaders < 2 and sig_args['dataloader_idx'].default is not inspect.Parameter.empty:
```

might need to split the line though